superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -40,6 +46,29 @@ import { exploreChart } from '../../explore/exploreUtils';
 import { CtasEnum } from '../actions/sqlLab';
 import { Query } from '../types';
 
+const updateDatset = async (
+  datasetId: number,
+  dbId: number,
+  sql: string,
+  columns: Array<Record<string, any>>,
+  overrideColumns: boolean,
+) => {
+  const endpoint = `api/v1/dataset/${datasetId}?override_columns=${overrideColumns}`;
+  const headers = { 'Content-Type': 'application/json' };
+  const body = JSON.stringify({
+    sql,
+    columns,
+    database_id: dbId,
+  });
+
+  const data: JsonResponse = await SupersetClient.put({
```

Comment:
did you want to use makeApi here as well to be consistent? What did you think about using the hooks api?

Reply:
Was mostly scoping this PR for removing the datasets, then another ticket for converting this into a makeApi put. Let me know what you think, I could do the work here

Reply:
I'm guessing changing to makeApi should be pretty easy but to hooks api more complicated?

Reply:
+1 on migrating this to `makeApi` in this PR. It handles the content type for you.